arch/src/aarch64/mod.rs

```diff
@@ -134,7 +137,9 @@ pub fn configure_system<T: DeviceInfoForFdt + Clone + Debug, S: ::std::hash::Bui
         initrd,
         pci_space_address,
     )
-    .map_err(Error::SetupFdt)?;
+    .unwrap();
```

Comment:
@rbradford @michael2012z 

I know this `unwrap()` here is a little bit weird and therefore I mark this PR as draft for now. The reason is that current vm-fdt code (https://github.com/rust-vmm/vm-fdt/blob/master/src/lib.rs#L11) does not expose the `vm_fdt::writer::Error` as public, so that we cannot use `vm_fdt::writer::Error` and pass the error using `?` in CLH.

What we are doing now is applying the legal for the `vm-fdt` crate and @jongwu will submit a PR to expose `vm_fdt::writer::Error` after we are allowed to contribute to that crate.

Reply:
I don't object to you doing `.map_err(|| Error::SetupFdt)` for now.

Reply:
> I don't object to you doing `.map_err(|| Error::SetupFdt)` for now.

Hi @rbradford Thanks for the amazingly quick response ^^

Hope I am understanding correctly, I think then we need to call another `unwrap()` in `write_fdt_to_memory` as otherwise the compiler will complain
```rust
   --> arch/src/aarch64/mod.rs:142:30
    |
142 |     fdt::write_fdt_to_memory(fdt_final, guest_mem).map_err(Error::WriteFdtToMemory)?;
    |                              ^^^^^^^^^ expected struct `std::vec::Vec`, found enum `std::result::Result`
    |
    = note: expected struct `std::vec::Vec<_>`
                 found enum `std::result::Result<std::vec::Vec<_>, fn(aarch64::fdt::Error) -> aarch64::Error {aarch64::Error::SetupFdt}>`
```
and if we do this, we still will get
```rust
 --> arch/src/aarch64/mod.rs:142:40
    |
142 |     fdt::write_fdt_to_memory(fdt_final.unwrap(), guest_mem).map_err(Error::WriteFdtToMemory)?;
    |                                        ^^^^^^ method not found in `std::result::Result<std::vec::Vec<u8>, fn(aarch64::fdt::Error) -> aarch64::Error {aarch64::Error::SetupFdt}>`
    |
    = note: the method `unwrap` exists but the following trait bounds were not satisfied:
            `fn(aarch64::fdt::Error) -> aarch64::Error {aarch64::Error::SetupFdt}: std::fmt::Debug`
```

To me that is a little bit too complex....if that `vm_fdt::writer::Error` is public, we can just create something like `FdtWriterError(vm_fdt::writer::Error)` in the `Error` of `fdt.rs` and change most of the function there to return `Result` instead of `FdtWriterError`. This keeps the error handling logic clean and simple.

Reply:
You might have to change the definition of `Error::SetupFdt` if it takes a parameter. Basically i'm suggesting you generate an error but without wrapping the upstream error.

Reply:
Ahh my bad, indeed you are absolutely right. I've updated the code. Thank you very much for the suggestion :))))